### DISCUSSION TITLE
testing t swap

### DISCUSSION THREAD

**Discussion by monaldmuck** [Cyfrin/security-and-auditing-full-course-s23#141] [Q&A] [answered]
when i use :pragma solidity 0.8.20; i get
Source file requires different compiler version (current compiler is 0.8.19+commit.7dd6d404.Emscripten.clang) - note that nightly builds are considered to be strictly less than the released version

when i change to: pragma solidity >=0.8.19 <0.9.0;

i get an error with my: import {Test} from "forge-std/Test.sol";
Source "forge-std/Test.sol" not found: File import callback not supported

please help

**Comment by alymurtazamemon** [comment]
@monaldmuck [Here](https://github.com/crytic/solc-select) is the documentation of the `solc-select` package which allows switching between solidity versions. Install the required version using it and see.

**Comment by EngrPips** [comment]
Hello @monaldmuck , did you figure it out already with the help of @alymurtazamemon  ?

**Reply by monaldmuck** [reply]
naw getting this now 
Warning (3420): Source file does not specify required compiler version! Consider adding "pragma solidity ^0.8.20;"
--> test/invariant-break/handler/Invariant.t.sol

Error (2904): Declaration "stdInvariant" not found in "lib/forge-std/src/StdInvariant.sol" (referenced as "forge-std/StdInvariant.sol").
 --> test/invariant-break/handler/AttemptedBreakTest.t.sol:6:1:
  |
6 | import {stdInvariant} from "forge-std/StdInvariant.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

**Reply by alymurtazamemon** [reply, marked as answer]
@monaldmuck The file is not compatible with your required solidity version. Check foundry.toml file which version you have specified there and make sure every file is compatible according to it.

**Reply by monaldmuck** [reply]
[compiler]
solidity_version = 0.8.20 i am updating the foundry.toml file
and when running fuzz test getting:
**Error: 
failed to extract foundry config:
foundry.toml error: TOML parse error at line 32, column 23
   |
32 | solidity_version = 0.8.20
   |                       ^
expected newline, `#`
 in foundry.toml TOML file**

**Reply by alymurtazamemon** [reply]
@monaldmuck it is `solc` 

```toml
solc = "0.8.20"
```

**Reply by monaldmuck** [reply]
thanks, I just figured it out, that helped & `import { StdInvariant } from "forge-std/StdInvariant.sol";`

I appreciate all the support

**Reply by EngrPips** [reply]
Glad you figured it buddy, keep crushing it!
